src/shared/sam/cli/samCliBuild.ts

```diff
@@ -35,9 +35,10 @@ export class SamCliBuildInvocation {
         console.error('SAM CLI error')
         console.error(`Exit code: ${exitCode}`)
         console.error(`Error: ${error}`)
+        console.error(`stderr: ${stderr}`)
         console.error(`stdout: ${stdout}`)
 
-        throw new Error(`sam build encountered an error: ${error && error.message ? error.message : stdout}`)
+        throw new Error(`sam build encountered an error: ${error && error.message || stderr || stdout}`)
```

Comment:
`error && error.message` evaluates to a `boolean`, so if neither are falsey, the message will be:

    sam build encountered an error: true